### PR TITLE
Add ability to change licence to answer edition

### PR DESCRIPTION
**Pull request by binaryberry**
The `build_clone` method from the edition model now handles
conversions from LicenceEdition to AnswerEdition.

cc @benilovj @boffbowsh

### CODE REVIEW

File: test/models/edition_test.rb [hunk 1]
```diff
@@ -222,6 +222,30 @@ def draft_second_edition_from(published_edition)
...
+      licence_overview: "I am a test overview",
+      licence_identifier: "Test identifier",
+      licence_short_description: "I am a test short description",
+      will_continue_on: "test will continue on",
+      continuation_link: "https://github.com/alphagov/"
+    )
+    new_edition = edition.build_clone AnswerEdition
+
+    assert_equal new_edition.class, AnswerEdition
+    assert_equal new_edition.version_number, 2
```
Comment (benilovj) [inline test/models/edition_test.rb]: it's best to reverse the parameters here, having "expected" as the first parameter and "actual" as the second parameter (see the [APIdocs](http://apidock.com/ruby/Test/Unit/Assertions/assert_equal)). This applies to other lines in the test.

Comment (jamiecobbett) [inline test/models/edition_test.rb]: :+1: the biggest benefit being that the failure messages will be correct instead of confusing.

Comment (binaryberry) [inline test/models/edition_test.rb]: Thanks! Amended now, will do a separate PR to amend the order in the other neighbouring tests.
